src/runtime_src/core/tools/common/ReportMechanical.cpp

```diff
@@ -57,12 +34,13 @@ void
 ReportMechanical::getPropertyTree20202( const xrt_core::device * _pDevice, 
                                            boost::property_tree::ptree &_pt) const
 {
-  boost::property_tree::ptree fan_array;
+  auto device = xrt::device(_pDevice->get_device_id());

```

Comment:
Looking at this constructor call, it is apparent we need a constructor xrt::device(const xrt_core::device*), because the code here is going over the bridge after the water, it is constructing a new device calling xclOpen, which is slow and not needed.

Lacking the constructor, please change to:
`xrt::device device(_pDevice->get_device_handle());`
while still not great, at least this avoids an xclOpen call.

Reply:
Done